src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
+const MAX_HEALTH = 100;
+const BITE_DAMAGE = 50;
+const DEAD_VALUE = 0;
+
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.name = name;
+    this.health = MAX_HEALTH;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
+      animal.health -= BITE_DAMAGE;
+      Animal.alive = Animal.alive.filter((item) => item.health > DEAD_VALUE);
+    }
+  }
```

Comment:
The `bite` method should also check if the target animal is not a Carnivore. Currently, it only checks if the target is a Herbivore and not hidden. Add an additional condition to verify that the target is not an instance of Carnivore.